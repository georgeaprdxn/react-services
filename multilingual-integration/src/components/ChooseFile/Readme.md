ChooseFile

```jsx inside Markdown
import mockData from './__test__/ChooseFile.mock.json'
const changeHandler = () => setPopup()

;<ChooseFile
	name={mockData.name}
	accept={mockData.accept}
	handleChange={changeHandler}
/>
```
