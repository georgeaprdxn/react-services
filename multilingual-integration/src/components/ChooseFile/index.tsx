import React, { FunctionComponent } from 'react'

import { StyledInput } from './styled'

type Props = {
	/**
	 * name of the input field
	 */
	name?: string
	/**
	 * acceptabe file type
	 */
	accept?: string
	/**
	 * handles change event
	 */
	handleChange?: React.FormEventHandler<HTMLInputElement>
}

const ChooseFile: FunctionComponent<Props> = ({
	name,
	accept,
	handleChange,
}) => {
	return (
		<StyledInput
			type="file"
			name={name}
			id={name}
			accept={accept}
			onChange={handleChange}
			className="custom-file-input"
		/>
	)
}

ChooseFile.defaultProps = {
	name: 'upload file',
	accept: '.jpg, .jpeg, .png',
}

export default ChooseFile
