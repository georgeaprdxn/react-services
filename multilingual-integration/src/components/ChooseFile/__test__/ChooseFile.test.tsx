import React from 'react'
import { shallow } from 'enzyme'

import ChooseFile from '../index'
import mockChoosFileData from './ChooseFile.mock.json'

describe('Choose file component tests', () => {
	let ChooseFileComp: any = null
	const handleChangeStub = jest.fn()
	beforeEach(() => {
		ChooseFileComp = shallow(
			<ChooseFile
				name={mockChoosFileData.name}
				accept={mockChoosFileData.accept}
				handleChange={handleChangeStub}
			/>
		)
	})

	it('should render a <input type="file"/>', () => {
		expect(ChooseFileComp.find('.custom-file-input').length).toEqual(1)
	})

	it('validates all props', () => {
		expect(ChooseFileComp.find('.custom-file-input').prop('name')).toEqual(
			mockChoosFileData.name
		)
		expect(ChooseFileComp.find('.custom-file-input').prop('type')).toEqual(
			'file'
		)
		expect(ChooseFileComp.find('.custom-file-input').prop('id')).toEqual(
			mockChoosFileData.name
		)
		expect(ChooseFileComp.find('.custom-file-input').prop('accept')).toEqual(
			mockChoosFileData.accept
		)
	})

	it('matches the snaphot', () => {
		expect(ChooseFileComp).toMatchSnapshot()
	})
})
