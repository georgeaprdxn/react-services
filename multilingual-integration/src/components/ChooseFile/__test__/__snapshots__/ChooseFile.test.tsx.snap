// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choose file component tests matches the snaphot 1`] = `
<styled.input
  accept=".jpg, .jpeg, .png"
  className="custom-file-input"
  id="file"
  name="file"
  onChange={[MockFunction]}
  type="file"
/>
`;
